app/src/main/java/org/mozilla/focus/activity/MainMediator.java

```diff
@@ -105,10 +105,12 @@ private void clearAllFragmentImmediate() {
     }
 
     public void dismissUrlInput() {
-        final Fragment top = getTopFragment();
-        if (UrlInputFragment.FRAGMENT_TAG.equals(top.getTag())) {
-            this.activity.onBackPressed();
-        }
+        final FragmentManager fragmentMgr = this.activity.getSupportFragmentManager();
+        final Fragment urlInputFrg = fragmentMgr.findFragmentByTag(UrlInputFragment.FRAGMENT_TAG);
+        // If UrlInputFragment exists, remove it and clear its transaction from back stack
+        FragmentTransaction clear = fragmentMgr.beginTransaction();
+        clear = (urlInputFrg == null) ? clear : clear.remove(urlInputFrg);
+        clear.commit();
```

Comment:
In method `MainMediator.showUrlInput()` we will add UrlInputFragment into back-stack. Per original implementation, back key will remove this fragment from back-stack.

In this commit, it only remove a fragment, and I guess the transaction will persist in back-stack. To around this, `FragmentManager.popBackStackImmediate` might be helpful.